Mon Mar 13 00:53:59 EST 2006  David M. Cooke <cookedm@physics.mcmaster.ca>
  * [03-warnings-in-utility-routines] Correct some warnings arising from utility routines
diff -rN -u Pyrex/Compiler/ExprNodes.py Pyrex/Compiler/ExprNodes.py
--- Pyrex/Compiler/ExprNodes.py	2006-03-13 03:34:20.000000000 -0500
+++ Pyrex/Compiler/ExprNodes.py	2006-03-13 03:34:20.000000000 -0500
@@ -3083,23 +3083,23 @@
 }
 
 static PyObject *__Pyx_UnpackItem(PyObject *seq, int i) {
-  PyObject *item;
-  if (!(item = PySequence_GetItem(seq, i))) {
+  PyObject *item = PySequence_GetItem(seq, i);
+  if (!item) {
     if (PyErr_ExceptionMatches(PyExc_IndexError))
-    	__Pyx_UnpackError();
+      __Pyx_UnpackError();
   }
   return item;
 }
 
 static int __Pyx_EndUnpack(PyObject *seq, int i) {
-  PyObject *item;
-  if (item = PySequence_GetItem(seq, i)) {
+  PyObject *item = PySequence_GetItem(seq, i);
+  if (item) {
     Py_DECREF(item);
     __Pyx_UnpackError();
     return -1;
   }
   PyErr_Clear();
-    return 0;
+  return 0;
 }
 """
 
diff -rN -u Pyrex/Compiler/Nodes.py Pyrex/Compiler/Nodes.py
--- Pyrex/Compiler/Nodes.py	2006-03-13 03:34:20.000000000 -0500
+++ Pyrex/Compiler/Nodes.py	2006-03-13 03:34:20.000000000 -0500
@@ -3762,10 +3762,8 @@
 bad:
     Py_XDECREF(args1);
     Py_XDECREF(kwds1);
-    if (*args2)
-        Py_XDECREF(*args2);
-    if (*kwds2)
-        Py_XDECREF(*kwds2);
+    Py_XDECREF(*args2);
+    Py_XDECREF(*kwds2);
     return -1;
 }
 """

